src/libPMacc/include/nvidia/reduce/Reduce.hpp

```diff
@@ -57,38 +56,39 @@ namespace PMacc
                     /* create a pointer with the right type*/
                     Type* s_mem=(Type*)s_mem_extern;
 
-                    if (tid >= src_count) return; /*end not needed threads*/
+                    if (tid >= src_count)
+                        return; /*end not needed threads*/
 
                     /*fill shared mem*/
                     Type r_value = src[tid];
-                    /*reduce not readed global memory to shared*/
+                    /*reduce not read global memory to shared*/
                     uint32_t i = tid + globalThreadCount;
                     while (i < src_count)
                     {
                         func(r_value, src[i]);
                         i += globalThreadCount;
                     }
-                    s_mem[l_tid] = r_value;
+                    s_mem[threadIdx.x] = r_value;
```

Comment:
I'd say threadIdx.x is more clear in the intention especially with the "elimination" of the threads. Besides that `blockIdx.x * blockDim.x + threadIdx.x` is a so common pattern in CUDA that it is easier to recognize than the one with `l_tid`